src/core/types.ml

```diff
@@ -11,3 +11,11 @@ type user_data =
   ]
 (** User defined data, generally passed as key/value pairs to
     whatever collector is installed (if any). *)
+
+type explicit_span = {
+  span: span;
+      (** Identifier for this span. Several explicit spans might share the same
+      identifier since we can differentiate between them via [meta]. *)
+  meta: Meta_map.t;  (** Metadata for this span (and its context) *)
+}
```

Comment:
do we want this to be public/constructable? perhaps it should be opaque to both collectors and library-authors, so that we can control the impl?

Reply:
At least in `trace-tef` it's already used to get/set data in `meta` (to transmit that data from the enter to the exit, basically). We need to at least expose get/set meta functions if we make this private.

On the face of it I do like the idea of making this private though.

Reply:
Would it be possible to lock it down entirely, and expose _specifically_ what `tef` needs; then leave it as a 'talk to us if you need us to expose more `meta` interfaces to collectors' kinda thing?

I'm just concerned about providing too much, too early, and then being locked out of future changes — backward- and forward-compatibility are _extremely_ important for this project, for obvious reasons.

Reply:
I'm worried that, a few years down the line, we have dozens (hundreds?) of bits of metadata exposed here because `trace` becomes a chokepoint for all communications between the user and the collector.